AM/PM Merged, Updated, Fixed

rdev-software opened the pull request:
Adding AM/PM feature to time picker and datetime picker
https://github.com/dylanwuzh/flutter-cupertino-date-picker/issues/35

Used code from PR https://github.com/dylanwuzh/flutter-cupertino-date-picker/pull/49
Retested, Added missing features and fixed some bugs with datetime picker

JamesMcIntosh committed:
AMPM

radvansky-tomas committed:
Resolve bugs with rendering date-time picker

Merge branch 'ampm' of https://github.com/JamesMcIntosh/flutter-cupertino-date-picker

# Conflicts:
#	lib/src/widget/datetime_picker_widget.dart
#	lib/src/widget/time_picker_widget.dart

radvansky-tomas committed:
Fix AM/PM selection

jamesdixon commented:
@dylanwuzh can we get this merged? 👍

jamesdixon commented:
@rdev-software I'm seeing an issue where when I attempt to scroll any of the wheels, the hours disappear. See video here: https://share.getcloudapp.com/nOumbNvl

Here's the code I'm using:

```dart
DatePicker.showDatePicker(
          context,
          initialDateTime: initialValue,
          dateFormat: 'MMM dd, yyyy|hh|mm|a',
          minDateTime: DateTime.now(),
          pickerMode: DateTimePickerMode.datetime,
          onConfirm: (v, _) {
            selectedDateTime = v;
            controller.text = formatDateTime(v);
            Vibrate.feedback(FeedbackType.success);
          },
        );
```

Are you experiencing this?

jamesdixon commented:
If I change the `dateFormat` to use 24 hour (`HH`), it works as expected.

carlleeswanson commented:
Would love to see this merged and available soon. Thanks for the enhancement!

Commit:
Merge branch 'master' of https://github.com/dylanwuzh/flutter-cupertino-date-picker into ampm

# Conflicts:
#	lib/src/widget/datetime_picker_widget.dart

Commit:
Fixed merge issue, add AM/PM to example

hinterlandcreative commented:
@dylanwuzh Can this get merged?

rdev-software committed:
Add missing setter for _currAmpm value

rdev-software committed:
Make widget state responsive

rdev-software committed:
Fix initial value of scroll views

rdev-software committed:
Remove DiagnosticableMixin

This removes DiagnosticableMixin, since it is a duplicate implementation of Diagnosticable. It existed in order to support plugins that wanted to support both master and stable, but now Diagnosticable is a mixin on stable, so we don't need it anymore.

rjtubera commented:
Hi is this implemented in the production?
